core/src/main/java/io/grpc/internal/GrpcUtil.java

```diff
@@ -131,9 +131,9 @@
   public static final String MESSAGE_ACCEPT_ENCODING = "grpc-accept-encoding";
 
   /**
-   * The default maximum uncompressed size (in bytes) for inbound messages. Defaults to 100 MiB.
+   * The default maximum uncompressed size (in bytes) for inbound messages. Defaults to 4 MiB.
    */
-  public static final int DEFAULT_MAX_MESSAGE_SIZE = 100 * 1024 * 1024;
+  public static final int DEFAULT_MAX_MESSAGE_SIZE = 4 * 1024 * 1024;
```

Comment:
If you don't have your heart set on 4194304, you might consider using 4000000 instead since it makes for easier to understand error messages.